src/backend/aspen/workflows/pangolin/find_samples.py

```diff
@@ -38,14 +38,14 @@ def find_samples():
         # TODO: update this comparison to be <= most_recent_pango_version
         # once we update this field to be a date instead of string
 
-        samples_to_be_updated: Iterable[str] = [
-            s.public_identifier
-            for s in all_samples
-            if (s.uploaded_pathogen_genome.pangolin_version is None)
-            or (
-                s.uploaded_pathogen_genome.pangolin_version != most_recent_pango_version
-            )
-        ]
+        samples_to_be_updated: Iterable[str] = []
+        for s in all_samples:
+            if s.uploaded_pathogen_genome is not None:
+                if (s.uploaded_pathogen_genome.pangolin_version is None) or (
+                    s.uploaded_pathogen_genome.pangolin_version
+                    != most_recent_pango_version
+                ):
+                    samples_to_be_updated.append(s.public_identifier)
```

Comment:
```suggestion
        samples_to_be_updated: Iterable[str] = [
            sample.public_identifier
            for sample in all_samples
            if (
                sample.uploaded_pathogen_genome is not None
                and sample.uploaded_pathogen_genome.pangolin_version != most_recent_pango_version
            )
        ]
```

Reply:
Actually, what you should do is to filter this above in your initial query.  No need to dump data if you're not going to use it.